subprojects/language-jvm/src/main/java/org/gradle/api/internal/tasks/compile/daemon/AbstractDaemonCompiler.java

```diff
@@ -47,7 +51,7 @@ public AbstractDaemonCompiler(File daemonWorkingDir, Compiler<T> delegate, Worke
     public WorkResult execute(T spec) {
         DaemonForkOptions daemonForkOptions = toDaemonOptions(spec);
         Worker<WorkerCompileSpec<?>> worker = workerFactory.getWorker(CompilerDaemonServer.class, daemonWorkingDir, daemonForkOptions);
-        DefaultWorkResult result = worker.execute(new WorkerCompileSpec<T>(delegate, spec));
+        DefaultWorkResult result = worker.execute(new WorkerCompileSpec<T>(delegate, spec, OperationIdentifierRegistry.getCurrentOperationId()));
```

Comment:
@adammurdoch I may be missing something here. What I'm trying to do is transfer the `OperationIdentifier` (and later the `BuildOperation#getId()`) to the worker process. The `WorkerSpec` gets serialized and send to the worker. Would you suggest to hook at a lower level and transfer the ids before transferring the `WorkerSpec` as a separate step?